### DISCUSSION TITLE
Preparations (in addition to cu* DLLs) for running on CUDA

### DISCUSSION THREAD

**Discussion by MMasutin** [Purfview/whisper-standalone-win#196] [Q&A] [answered]
Just in case, I selected to post a question, not an issue.
I have so far failed to do it on different Nvidia GPUs, both older and quite recent. I don't understand versions required, i.e. compatibility issues with drivers and other files. Are there useful tips?
2) If an additional question is possible here, is there a variable to set 'minimum duration of pause/silence' that would mark a next segment?

**Comment by Purfview** [comment]
You didn't posted what is the issue with "CUDA". You can use libs from there -> https://github.com/Purfview/whisper-standalone-win/releases/tag/libs

>2) If an additional question is possible here, is there a variable to set 'minimum duration of pause/silence' that would mark a next segment?

Use `--sentence`, for it there is sub-parameter `--max_gap`. It's 3.0 seconds by default.

**Comment by MMasutin** [comment]
Thanks for your reply! The issue is how to make the CUDA capabilities available, in general. Not an issue with the program but a question of how to ensure that the Nvidia GPU is recognized correctly. I know about the libs. It's not enough to add them, and I don't understand what other steps are required. Must the versions of some files be compatible? Is there a kind of check list? For example, I check the version of the Nvidia driver and then what?

**Comment by Purfview** [comment]
>Not an issue with the program

Then I'm converting this to the discussion.

>ensure that the Nvidia GPU is recognized correctly

I think, it's recognized or not, if not then post what error you get.

**Reply by hnorgaar** [reply]
Best way is to to add the CUDA files to Windows/system32, else you need to add the path to Windows enviroment variables for where they are stored

**Comment by MMasutin** [comment]
It doesn't help having the cu* DLLs in %WinDir%\System32 (What about a 64-bit system?) or a folder in the %Path% variable. Using --checkcuda returns 0. Tried on two GPUs. And the errors, when trying to run a trascription job on CUDA, may be typical.
One of the GPUs I tried has the driver version 353.62. The other pieces of NV software may need to correspond to this. If CUDA doesn't work by simply putting the DLLs along with Whisper-Faster, then I don't understand the whole process of finding all the pieces of compatible software.
`Traceback (most recent call last):
File "D:\whisper-fast\__main__.py", line 1104, in <module>
File "D:\whisper-fast\__main__.py", line 972, in cli
File "faster_whisper\transcribe.py", line 136, in __init__
RuntimeError: CUDA failed with error initialization error
[4084] Failed to execute script '__main__' due to unhandled exception!`

**Reply by Purfview** [reply, marked as answer]
> Using --checkcuda returns 0. Tried on two GPUs.

What GPUs? NVIDIA GPUs should have Compute Capability greater or equal to 3.5.

> One of the GPUs I tried has the driver version 353.62

That version doesn't support CUDA 11, it was introduced with 452.39, so you need that or newer driver.

**Comment by MMasutin** [comment]
Thanks for your replies!
Thus prerequisites are:
CUDA Capability 3.5 or greater (according to [this](https://docs.nvidia.com/datacenter/tesla/drivers/#software-matrix), present in NV family Kepler or newer);
Support of CUDA Toolkit 11.x, i.e. the driver version 452.39 or newer.
How do I best go about verifying if these can be met?
First, I probably do a search for the GPU model to see if it's a compatible architecture (at least Kepler).
Then, where or how do I search for a new driver? It's not what comes with an installation pack for a given GPU?

**Reply by Purfview** [reply]
> How do I best go about verifying if these can be met?

Maybe with tool like Aida64.

> Then, where or how do I search for a new driver?

Official: https://www.nvidia.com/Download/Find.aspx?lang=en-us
Here you can find info how to mod drivers: https://forums.guru3d.com/forums/videocards-nvidia-geforce-drivers-section.21/

**Reply by 1001ruchka** [reply]
GPU-Z
https://www.techpowerup.com/gpuz/

![image](https://github.com/Purfview/whisper-standalone-win/assets/5934321/e3651f7d-838f-4bac-aa07-3cb6a7df9e39)
